pow-migration: Fix test compilation

jsdanielh opened the pull request:
- Fix compilation of tests in the `pow-migration` subcrate. This fixes #2680.
- Check for compilation errors in tests with all features enabled and benchmarks. This was added as a step into the `build+test` workflow in GH Actions.


## Pull request checklist

- [X] All tests pass. The project builds and runs.
- [X] I have resolved any merge conflicts.
- [X] I have resolved all `clippy` and `rustfmt` warnings.

dependabot[bot] committed:
build(deps): bump docker/build-push-action

Bumps the minor-and-patch-updates group with 1 update: [docker/build-push-action](https://github.com/docker/build-push-action).


Updates `docker/build-push-action` from 6.1.0 to 6.2.0
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/31159d49c0d4756269a0940a750801a1ea5d7003...15560696de535e4014efeff63c48f16952e52dd1)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch-updates
...

Signed-off-by: dependabot[bot] <support@github.com>

sisou committed:
Use PoW memory cost for Argon2d KDF

This makes the function backward-compatible with the PoW JS library, to support decoding existing LoginFiles.

Partly reverses 9f8db328a368c689bb5a1366f92d39da52e5d2f4 (#2653) and changes it to be in the spirit of #2377, which meant to preserve backward-compatibility.

jsdanielh committed:
pow-migration: Fix test compilation

Fix compilation of tests in the `pow-migration` subcrate.
This fixes #2680.

hrxi reviewed (commented):
The code looks good to me. Could you make sure that this test won't fail to compile in the future? Maybe we could run a `cargo check --all-features --tests --benches` in CI?

Eligioo committed:
Fix LowBlockProducerTimeout devnet scenario

jsdanielh committed:
ci: Check for compilation errors in tests and benchmarks

Check for compilation errors in tests with all features enabled and
benchmarks. This was added as a step into the `build+test` workflow
in GH Actions.

jsdanielh commented:
> The code looks good to me. Could you make sure that this test won't fail to compile in the future? Maybe we could run a `cargo check --all-features --tests --benches` in CI?

Done.

hrxi commented on .github/workflows/build+test.yml:
```diff
@@ -20,6 +20,14 @@ jobs:
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@stable
```
```suggestion
    - uses: dtolnay/rust-toolchain@stable
    - uses: Swatinem/rust-cache@v2
```
Actually, can you add the cache here?
